bin/generate-bundle.js

```diff
@@ -0,0 +1,233 @@
+ */
+class CssProcesser {
+  constructor(dsV, args) {
+    this.dsV = dsV;
+    this.args = args;
+    this.processed = [];
+    this.skipped = [];
+
+    let classDef = '';
+    if (args.scopeClass) {
+      for (let i = 0; i < args.scopeSpecificity; i++) {
+        classDef += `.${args.scopeClass}`;
+      }
+    }
+    this.classDef = classDef;
+  };
+
+  run() {
+    return this.generateLESS()
+      .then((raw) => this.compileLess(raw, autoprefixPlugin))
+      .then((raw) => this.compileLess(raw, cleanCSSPlugin))
+      .then((raw) => this.writeAllFiles(raw))
+      .catch((e) => console.error(e));
+  }
+
+  getDistCss() {
+    return new Promise((resolve, reject) => {
+      glob(`${currentDir}/dist/**/${this.dsV}/**/*.css`, (err, files) => {
+        if (err) {
+          return reject(err);
+        }
+        resolve(files);
+      });
+    });
+  }
+
+  /**
+   * Generates a raw less file for given css file
+   * @param {*} file
+   */
+  generateRawLess(file) {
+    const fileName = path.basename(file, '.css');
+    if (this.args.modules.length > 0 && this.args.modules.indexOf(fileName) === -1) {
+      this.skipped.push(fileName);
+      return '';
+    }
+    this.processed.push(fileName);
+
+    if (!!this.classDef) {
+      // Need to read file and output
+      const cssContents = fs.readFileSync(file);
+      return `${this.classDef} {
+             ${cssContents}
+            }`;
+    } else {
+      return `@import (inline) "${file}";`;
+    }
+  }
+
+  processFiles(files) {
+    return new Promise((resolve, reject) => {
+      const compiled = files.map((file) => this.generateRawLess(file)).join('\n');
+      const processed = this.processed
+      const skipped = this.skipped;
+      const dsV = this.dsV;
+
+      if (this.args.modules.length > 0 || this.args.verbose) {
+        console.log(`Processed ${processed.length} modules for ${dsV}`)
+      }
+      if (this.args.verbose) {
+        console.log(`Modules processed: ${processed.join(',')} for ${dsV}`)
+
+        console.log(`Skipped ${skipped.length} modules for ${dsV}`)
+        console.log(`Modules skipped: ${skipped.join(',')} for ${dsV}`)
+      }
+      resolve(compiled);
+    });
+  }
+
+  generateLESS() {
+    return this.getDistCss(this.dsV).then((files) => this.processFiles(files));
+  }
+
+  compileLess(raw, plugin) {
+    return less.render(raw, { plugins: [plugin] })
+      .then((render) => render.css,
+      );
+  }
+
+  writeAllFiles(raw) {
+    const path = getCDNPath(this.args.name, this.dsV);
+    return makeDir(path).then(() => writeFile(`${path}/skin.min.css`, raw));
+  }
+}
+
+function getCDNPath(bundle, dsV) {
+  return `${currentDir}/_cdn/${bundle}/v${pkg.version}/${dsV}`;
+}
+
+
+/**
+ * Runs the compilation of less
+ * @param {*} res The response with dsVersion and raw css/less
+ * @param {*} plugin The given plugin to run the render with
+ */
+/**
+ * Runs npm build to get dist output
+ */
+function prebuild() {
```

Comment:
Missing colons on both of the `console.log` calls.  Are these calls needed or were they just for debugging?

Reply:
Yes, they were for spacing out things, some console logs are used for verbose messaging.

Reply:
Can you make sure they end with `;` then.  Are we linting this file?

Reply:
We are not linting the file, we don't have any javascript linters installed in skin.

Reply:
duh... I thought I saw eslint config files, but I was looking at core.
OK.